cmsplugin_articles_ai/views.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+import json
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import DetailView, ListView
+
+from .models import Article, Tag, TagFilterMode
+
+
+class ArticleView(DetailView):
+    """
+    View for displaying single article.
+    """
+    model = Article
+    template_name = "cmsplugin_articles_ai/article_detail.html"
+    queryset = Article.objects.public()
+
+    def get_context_data(self, **kwargs):
+        context = super(ArticleView, self).get_context_data(**kwargs)
+        article = self.object
+        attachments = article.attachments.select_related("attachment_file")
+        context["image_attachments"] = attachments.images()
```

Comment:
Hrm. You end up doing two separate queries for attachments like this, one for images and one for non-images.

It would likely be faster to just fetch them in one go and sift them into two lists in the `get_context_data` function.